# Remote Management Request
temoto_2/RMPRequest rmp

# Type of the requested sensor (required).
string algorithm_type

# Name of the ROS package where algorithm is located. (optional)
# If the package name is left undefined
# then the first known entry under the specified type is executed
string package_name

# name of the executable (optional)
# When undefined, first defined entry is used.
string executable

# Input topics. Each element is a pair of strings where the 
# first element (key) describes the type of the topic and the
# second element (value) describes the name of the topic itself.
# Key is required but if the value is left empty, a default
# topic name is used (indicated by the algorithm info file)
# by the algorithm manager
diagnostic_msgs/KeyValue[] input_topics

# Same idea applies here as it did with the input_topics
diagnostic_msgs/KeyValue[] output_topics

---

# Remote Management Response
temoto_2/RMPResponse rmp

# name of the package
string package_name

# name of the executable/launchable/runnable
string executable

# Actual input topics
diagnostic_msgs/KeyValue[] input_topics

# Actual output topics
diagnostic_msgs/KeyValue[] output_topics
